handlers/holiday-stop-api/src/test/scala/com/gu/holiday_stops/HandlerTest.scala

```diff
@@ -69,64 +67,59 @@ class HandlerTest extends AnyFlatSpec with Matchers {
-        currentTerm = 12,
-        currentTermPeriodType = "Month",
-        autoRenew = true,
-        ratePlans = List(
-          RatePlan(
-            productName = "Guardian Weekly - Domestic",
-            ratePlanName = "GW Oct 18 - Quarterly - Domestic",
-            ratePlanCharges = List(
-              RatePlanCharge(
-                name = "GW Oct 18 - Quarterly - Domestic",
-                number = "C1",
-                37.50,
-                Some("Quarter"),
-                effectiveStartDate = startDate,
-                chargedThroughDate = Some(endDate),
-                HolidayStart__c = None,
-                HolidayEnd__c = None,
-                processedThroughDate = Some(endDate.minusMonths(3)),
-                "",
-                specificBillingPeriod = None,
-                endDateCondition = Some("Subscription_End"),
-                upToPeriodsType = None,
-                upToPeriods = None,
-                billingDay = None,
-                triggerEvent = Some("SpecificDate"),
-                triggerDate = Some(startDate),
-                discountPercentage = None,
-                effectiveEndDate = LocalDate.now,
+      inside(
+        unwrappedOp(
+          Handler.operationForEffects(
+            defaultTestEffects.response,
+            Stage("CODE"),
+            FakeFetchString.fetchString,
+            testBackend,
+            "test-generated-id",
+          ),
+        ).map { operation =>
+          operation
+            .steps(
+              legacyPotentialIssueDateRequest(
+                productPrefix = "Guardian Weekly xxx",
+                startDate = asIsoDateString(today.minusMonths(1)),
+                endDate = "2019-01-15",
+                subscriptionName = gwDomesticSubscription.subscriptionNumber,
+              ),
+            )
+        },
+      ) { case ContinueProcessing(response) =>
+        response.statusCode should equal("200")
+        val parsedResponseBody = Json.fromJson[PotentialHolidayStopsResponse](Json.parse(response.body))
+        inside(parsedResponseBody) { case JsSuccess(response, _) =>
+          response should equal(
+            PotentialHolidayStopsResponse(
+              nextInvoiceDateAfterToday = LocalDate.parse("2019-04-01"),
+              potentials = List(
+                PotentialHolidayStop(LocalDate.parse("2019-01-04"), Credit(-2.89, LocalDate.parse("2019-04-01"))),
+                PotentialHolidayStop(LocalDate.parse("2019-01-11"), Credit(-2.89, LocalDate.parse("2019-04-01"))),
               ),
             ),
-            productRatePlanId = "",
-            id = "",
-            lastChangeType = None,
-          ),
-        ),
-        "Active",
-        accountNumber = accountNumber,
-      )
+          )
+        }
+      }
+    }
+  "GET /potential/<<sub name>>?startDate=...&endDate=... endpoint" should
+    "calculate the correct amount for T3 subscriptions" in {
+      val today = LocalDate.parse("2024-06-20")
```

Comment:
very minor - could we call this fakeToday or referenceDate or similar
```suggestion
      val referenceDate = LocalDate.parse("2024-06-20")
```